Fix changing primary style name for NativeSelect

Artur- opened the pull request:
Depends on #9007

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/9016)
<!-- Reviewable:end -->

Artur- committed:
Fix changing primary style name for NativeSelect

tsuoanttila commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[client/src/main/java/com/vaadin/client/ui/VListSelect.java, line 62 at r1](https://reviewable.io:443/reviews/vaadin/framework/9016#-Kh1eAlaTexkO3H_VtAx:-Kh1eAlaTexkO3H_VtAy:b-94vz7e) ([raw file](https://github.com/vaadin/framework/blob/844374ea7a08f7d3fae369e7f35b15576be337ce/client/src/main/java/com/vaadin/client/ui/VListSelect.java#L62)):*
> ```Java
>     public VListSelect() {
>         container = new FlowPanel();
>         container.setStyleName(ListSelectState.PRIMARY_STYLENAME);
> ```

this style name is never updated in any case?

---

*[client/src/main/java/com/vaadin/client/ui/VListSelect.java, line 68 at r1](https://reviewable.io:443/reviews/vaadin/framework/9016#-Kh1eDj8AyrlcHIknYP9:-Kh1eDj8AyrlcHIknYPA:buaw6rr) ([raw file](https://github.com/vaadin/framework/blob/844374ea7a08f7d3fae369e7f35b15576be337ce/client/src/main/java/com/vaadin/client/ui/VListSelect.java#L68)):*
> ```Java
>         select = new ListBox();
>         select.setMultipleSelect(true);
>         select.setStyleName(NativeSelectState.STYLE_NAME + "-select");
> ```

why does this use a STYLE_NAME from NativeSelectState and not ListSelectState like container?

---

*[uitest/src/test/java/com/vaadin/tests/components/listselect/ListSelectStyleNamesTest.java, line 34 at r1](https://reviewable.io:443/reviews/vaadin/framework/9016#-Kh1ehO8jQbPnFMtLmVT:-Kh1ehO8jQbPnFMtLmVU:baqh67s) ([raw file](https://github.com/vaadin/framework/blob/844374ea7a08f7d3fae369e7f35b15576be337ce/uitest/src/test/java/com/vaadin/tests/components/listselect/ListSelectStyleNamesTest.java#L34)):*
> ```Java
> 
>     @Test
>     public void correctStyleNames() {
> ```

can this test be split into pieces? If something goes wrong in one step of this test, the rest of the test is not even executed. Further problems won't be known until first is fixed and test is re-run (which most people won't do when fixing a "trivial" issue).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9016)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[client/src/main/java/com/vaadin/client/ui/VListSelect.java, line 62 at r1](https://reviewable.io:443/reviews/vaadin/framework/9016#-Kh1eAlaTexkO3H_VtAx:-Kh3VKtYSVrVUj3nQs7U:bj9588m) ([raw file](https://github.com/vaadin/framework/blob/844374ea7a08f7d3fae369e7f35b15576be337ce/client/src/main/java/com/vaadin/client/ui/VListSelect.java#L62)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

this style name is never updated in any case?
</blockquote></details>

It's updated whenever the primary style name is changed. Changed to call `setStylePrimaryName` instead (in #9007 where the code really is) to be clearer

---

*[client/src/main/java/com/vaadin/client/ui/VListSelect.java, line 68 at r1](https://reviewable.io:443/reviews/vaadin/framework/9016#-Kh1eDj8AyrlcHIknYP9:-Kh3VMURGy3V7DlHosgD:b-gxuzhf) ([raw file](https://github.com/vaadin/framework/blob/844374ea7a08f7d3fae369e7f35b15576be337ce/client/src/main/java/com/vaadin/client/ui/VListSelect.java#L68)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

why does this use a STYLE_NAME from NativeSelectState and not ListSelectState like container?
</blockquote></details>

Fixed in #9007 where the code really is

---

*[uitest/src/test/java/com/vaadin/tests/components/listselect/ListSelectStyleNamesTest.java, line 34 at r1](https://reviewable.io:443/reviews/vaadin/framework/9016#-Kh1ehO8jQbPnFMtLmVT:-Kh3WQMn1RQIBuGIU1Lx:bqvl2fp) ([raw file](https://github.com/vaadin/framework/blob/844374ea7a08f7d3fae369e7f35b15576be337ce/uitest/src/test/java/com/vaadin/tests/components/listselect/ListSelectStyleNamesTest.java#L34)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

can this test be split into pieces? If something goes wrong in one step of this test, the rest of the test is not even executed. Further problems won't be known until first is fixed and test is re-run (which most people won't do when fixing a "trivial" issue).
</blockquote></details>

Done in #9007 where the code really is

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9016)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/9016#-:-Ki9qWXBgJlE0DPZQfh8:bnfp4nl)*
<!-- Sent from Reviewable.io -->
